### ISSUE TITLE
Datasets on reproducing FID results

### ISSUE THREAD

**Issue by senguet** [mlpc-ucsd/TokenCompose#9]
Thank you for your contribution! 
        I am unable to reproduce the FID results. 
        Can you provide 10000 image-caption pairs sampled from COCO validation set (C) and 1000 image-caption pairs from Flickr30K entities validation set (F)? 
        Thanks!

**Comment by JamesSand** [comment]
Hi. Thank you for your interest. For the COCO FID, we followed the evaluation method used by huggingface Stable Diffuision1.4. You can check on this link https://huggingface.co/CompVis/stable-diffusion-v1-4 . Namely, they use 10k random prompts from COCO 2017 Validation set. 
![image](https://github.com/mlpc-ucsd/TokenCompose/assets/98897204/0587b084-e8b6-4a20-b724-ad377e0bff84)

**Comment by senguet** [comment]
> 你好。感谢您的关注。对于COCO FID，我们遵循huggingface Stable Diffuision1.4使用的评估方法。您可以在此链接 https://huggingface.co/CompVis/stable-diffusion-v1-4 查看。也就是说，他们使用来自 COCO 2017 验证集的 10k 随机提示。 ![image](https://private-user-images.githubusercontent.com/98897204/345076496-0587b084-e8b6-4a20-b724-ad377e0bff84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NzI0ODMsIm5iZiI6MTcxOTk3MjE4MywicGF0aCI6Ii85ODg5NzIwNC8zNDUwNzY0OTYtMDU4N2IwODQtZThiNi00YTIwLWI3MjQtYWQzNzdlMGJmZjg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDAyMDMwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0ODQ5MWJlNDJjNzc2ZTcwZTgxNDEyOGZiM2RmNWJmNGNlMzgxMGM5YmRmZDFjZTlkMjc5MTJlNmVhNjQ1NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._9XiAwxKeNFWOdZInxEB3dfmAie8RqLlRJXiMsrjrug)


Thank you for your reply！
Perhaps I was wrong, but why did I find out that Coco 2017 val set only has 5k prompts.

**Comment by zwcolin** [comment]
There are 5k images in coco 17 val set but around 25k captions. Prompts are defined at the caption-level, not the image-level.

[val2017_rand_index.json](https://github.com/user-attachments/files/16076623/val2017_rand_index.json)

Feel free to download the above indices file if you want to ensure a fair comparison, though the variance should be very small even if you don't use exactly these indices. We center crop the images to ensure that they all have a square shape and use this [codebase](https://github.com/mseitzer/pytorch-fid/tree/master/src/pytorch_fid) to calculate the FID.

The same pipeline also applies for the Flickr30K entities (val set).

Let us know if you have any other questions -- thanks!

**Comment by senguet** [comment]
> coco 17 val 套装中有 5k 图像，但大约有 25k 字幕。提示是在标题级别定义的，而不是在图像级别定义的。
> 
> [val2017_rand_index.json](https://github.com/user-attachments/files/16076623/val2017_rand_index.json)
> 
> 如果您想确保公平的比较，请随时下载上述索引文件，尽管即使您不完全使用这些索引，方差也应该非常小。我们将图像居中裁剪，以确保它们都具有正方形，并使用此[代码库](https://github.com/mseitzer/pytorch-fid/tree/master/src/pytorch_fid)来计算 FID。
> 
> 同样的管道也适用于 Flickr30K 实体（val 集）。
> 
> 如果您有任何其他问题，请告诉我们 - 谢谢！

Thank you for your reply, it has been of great help to me. 
I have another question, how the efficiency indicator was obtained, especially how the standard deviation of efficiency was obtained.
 Look forward to your reply!

**Comment by zwcolin** [comment]
You can generate a number of images (say 100) with the same set of prompts for all your baselines using the same GPU. Then you time it and take the avg/std for efficiency metrics. We strongly recommend that you manually do this for all your baselines (instead of taking our numbers) for this column because latency can differ even for the same GPU due to different system configs, hardware brand, software, etc.

**Comment by senguet** [comment]
> You can generate a number of images (say 100) with the same set of prompts for all your baselines using the same GPU. Then you time it and take the avg/std for efficiency metrics. We strongly recommend that you manually do this for all your baselines (instead of taking our numbers) for this column because latency can differ even for the same GPU due to different system configs, hardware brand, software, etc.



Thank you for your reply, it has been of great help to me.
Wishing you a happy life.

**Comment by zwcolin** [comment]
I assume that all your questions have been resolved as of now, so I'm closing this issue. If you encounter any other questions, feel free to re-open it or submit a new issue. Have a great rest of your day!
